### PR TITLE
Add missing tests for GitResource

### PR DESCRIPTION
**Pull request by shuheiktgw**
# Changes

I just added some missing tests for `GitResource`. Thanks for your review! 😄 

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [x] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/pipeline/blob/master/CONTRIBUTING.md) for more details._

Double check this list of stuff that's easy to miss:

- If you are adding [a new binary/image to the `cmd` dir](../cmd), please update
  [the release Task](../tekton/publish.yaml) and [TaskRun](../tekton/publish-run.yaml) to build and release this image

### CODE REVIEW

File: pkg/apis/pipeline/v1alpha1/git_resource_test.go [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+		Revision:   "master",
+		TargetPath: "/test/test",
+	}
+
+	want := map[string]string{
+		"name":     "git-resource",
+		"type":     string(v1alpha1.PipelineResourceTypeGit),
+		"url":      "git@github.com:test/test.git",
+		"revision": "master",
+		"path":     "/test/test",
```
Comment (dibyom) [inline pkg/apis/pipeline/v1alpha1/git_resource_test.go]: Another test case here would be testing the `path` replacement when `TargetPath` is not specified. (Though we might be changing how that works based on #1115)

Comment (dlorenc) [reply pkg/apis/pipeline/v1alpha1/git_resource_test.go]: I think we decided to keep that around, so it should be safe to add.

**Pull request merged**